src/magic_folder/test/cli/test_magic_folder.py

```diff
@@ -164,51 +182,36 @@ def setUp(self):
         yield self.client_fixture.use_on(self)
 
         self.tempdir = self.client_fixture.tempdir
-        self.node_directory = self.client_fixture.node_directory
+        self.config_dir = FilePath(self.mktemp())
+        create_global_configuration(
+            self.config_dir,
+            u"tcp:4321",
+            self.client_fixture.node_directory,
+        )
 
     @defer.inlineCallbacks
-    def test_create_magic_folder(self):
+    def test_add_magic_folder(self):
         """
         Create a new magic folder with a nickname and local directory so
         that this folder is also invited and joined with the given nickname.
         """
         # Get a magic folder.
         magic_folder = self.tempdir.child(u"magic-folder")
+        magic_folder.makedirs()
+
         outcome = yield cli(
-            self.node_directory, [
-                b"create",
-                b"magik:",
-                b"test_create",
+            self.config_dir, [
+                b"add",
+                b"--name", b"test",
+                b"--author", b"test",
                 magic_folder.asBytesMode().path,
             ],
         )
-
         self.assertThat(
             outcome.succeeded(),
             Equals(True),
         )
 
-    @defer.inlineCallbacks
-    def test_create_error(self):
-        """
-        Try to create a magic folder with an invalid nickname and check if
-        this results in an error.
-        """
-        magic_folder = self.tempdir.child(u"magic-folder")
-        outcome = yield cli(
-            self.node_directory, [
-                b"create",
-                b"m a g i k",
-                b"test_create_error",
-                magic_folder.asBytesMode().path,
-            ]
-        )
-
-        self.assertThat(
-            outcome.succeeded(),
-            Equals(False),
-        )
-
```

Comment:
Shouldn't there be a similar test as `test_create_error()` for `magic-folder add`, or is that not necessary anymore?

(I tested out of curiosity: magic folder reported that my attempt to add a folder with a name that contains non-ASCII characters was successful, but `magic-folder list` balked at it.  I will report the issue separately.)

Reply:
Okay, yeah that definitely sounds like a bug.

This particular test is checking invalid chars in the alias (which is now gone) and spaces should be allowed in the nicknames (as well) anyway.

Reply:
A missing `unicode_literals` import fixes it .. not sure if that's "the strategy" we should use (versus `u""` everywhere) but I think that's what we decided to do in Tahoe for py3 port so..